app/components/Exchange/MyOpenOrders.jsx

```diff
@@ -558,133 +452,67 @@ class MyOpenOrders extends React.Component {
                 />
             );
 
-            footerContainer =
-                settleOrdersLength > 11 ? (
-                    <div className="orderbook-showall">
-                        <a onClick={this._onSetShowAll.bind(this)}>
-                            <Translate
-                                content={
-                                    showAll
-                                        ? "exchange.hide"
-                                        : "exchange.show_all_orders"
-                                }
-                                rowcount={settleOrdersLength}
-                            />
-                        </a>
-                    </div>
-                ) : null;
+            footerContainer = totalRows > 11 && (
+                <div className="orderbook-showall">
+                    <a onClick={this._onSetShowAll.bind(this)}>
+                        <Translate
+                            content={
+                                showAll
+                                    ? "exchange.hide"
+                                    : "exchange.show_all_orders"
+                            }
+                            rowcount={totalRows}
+                        />
+                    </a>
+                </div>
+            );
         }
 
+        let isSelected =
+            this.state.selectedOrders.length > 0 &&
+            this.state.selectedOrders.length == totalRows;
+
         return (
-            <div
+            <MarketsOrderView
```

Comment:
not a big fan of this props abundance

Reply:
We used these many props before, moving the code to its own merely visualize this. We could bundle the props and just set them on the component though.

Reply:
This seems to be an issue however. https://reactjs.org/docs/forwarding-refs.html

It doesn't have an issue on building locally, but it seems to be what breaks the current developer deployment?

https://github.com/bitshares/bitshares-ui/blob/42bf56bae0d13b7727c48c51d308b6e769a49c7d/app/components/Exchange/MyOpenOrders.jsx#L123

Reply:
Yea, the view is not a component and has refs inside. This also breaks the encapsulation imo.

Make the view a class (doesn't have to be a component) and give it a proper event interface with regards to what events are possible for view manipulation

Reply:
I've used this way, not really based on MVC separation recommendations, but for [this](https://programmingwithmosh.com/react/react-functional-components/). MVC separations can involve things like [Flux](https://github.com/facebook/flux).

The goal was to separate the HTML and SCSS code from the files that are getting very long at times. And since they didn't require, until this one, a class I was sticking to it. 

I can convert it to a class instead, and keep the new file structure, if you agree. I think it keeps the code easier to find in when the view parts are separated.